sky/execution.py

```diff
@@ -516,19 +517,38 @@ def spot_launch(
         common_utils.dump_yaml(f.name, task_config)
 
         controller_name = spot.SPOT_CONTROLLER_NAME
+        vars_to_fill = {
+            'remote_user_yaml_prefix': spot.SPOT_TASK_YAML_PREFIX,
+            'user_yaml_path': f.name,
+            'user_config_path': None,
+            'spot_controller': controller_name,
+            'cluster_name': name,
+            'gcloud_installation_commands': gcp.GCLOUD_INSTALLATION_COMMAND,
+            'is_dev': env_options.Options.IS_DEVELOPER.get(),
+            'disable_logging': env_options.Options.DISABLE_LOGGING.get(),
+            'logging_user_hash': common_utils.get_user_hash(),
+            'retry_until_up': retry_until_up,
+            'user': os.environ.get('USER', None),
+        }
+        if sky_config.loaded():
+            # Look up the contents of the already loaded configs via the
+            # 'sky_config' module. Don't simply read the on-disk file as it may
+            # have changed since this process started.
+            #
+            # Pop any proxy command, because the controller would've been
+            # launched behind the proxy, and in general any nodes we launch may
+            # not have or need the proxy setup.
+            config_dict = sky_config.pop_nested(('auth', 'ssh_proxy_command'))
+            with tempfile.NamedTemporaryFile(mode='w', delete=False) as tmpfile:
+                common_utils.dump_yaml(tmpfile.name, config_dict)
+                vars_to_fill.update({
+                    'user_config_path': tmpfile.name,
+                    'remote_user_config_path': sky_config.REMOTE_CONFIG_PATH,
```

Comment:
The following situation may be problematic as recovering the existing spot job will use the new `config` file uploaded by a later spot job:
1. `sky spot launch -n job1 sleep 10000`
2. change the config file
3. `sky spot launch -n job2 sleep 10000`
4. The job1 is now preempted and starts recovering. Since the recovering calls the `sky.launch` it will get the new config file?

Reply:
Nice! That makes a lot of sense! Thank you for the fix @concretevitamin.